On 0.4.1 WebhookClient cannot be used as a bean in Spring Boot

From the 0.4.1 update on, the `WebhookClient` cannot be used as a bean anymore in a Spring Boot application because it uses parameters of types of optional dependencies (other discord api libs) that are usually not included in the runtime classpath, so some Spring bean introspection fails.

I can write myself a wrapper fine, so it can be worked around easily, just thought you might want to know.

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'botStatusWebhookClient' defined in class path resource [space/npstr/aki/config/WebhookConfiguration.class]: Post-processing of merged bean definition failed; nested exception is java.lang.IllegalStateException: Failed to introspect Class [club.minnced.discord.webhook.WebhookClient] from ClassLoader [org.springframework.boot.loader.LaunchedURLClassLoader@4f2fe9c1]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:571) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:324) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$$Lambda$249/000000001102D280.getObject(Unknown Source) ~[na:na]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory$2.resolveCandidate(DefaultListableBeanFactory.java:1765) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1307) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.createOptionalDependency(DefaultListableBeanFactory.java:1768) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1214) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	... 42 common frames omitted
Caused by: java.lang.IllegalStateException: Failed to introspect Class [club.minnced.discord.webhook.WebhookClient] from ClassLoader [org.springframework.boot.loader.LaunchedURLClassLoader@4f2fe9c1]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:481) ~[spring-core-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.util.ReflectionUtils.doWithLocalMethods(ReflectionUtils.java:321) ~[spring-core-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.buildLifecycleMetadata(InitDestroyAnnotationBeanPostProcessor.java:232) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.findLifecycleMetadata(InitDestroyAnnotationBeanPostProcessor.java:210) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessMergedBeanDefinition(InitDestroyAnnotationBeanPostProcessor.java:149) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.context.annotation.CommonAnnotationBeanPostProcessor.postProcessMergedBeanDefinition(CommonAnnotationBeanPostProcessor.java:294) ~[spring-context-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyMergedBeanDefinitionPostProcessors(AbstractAutowireCapableBeanFactory.java:1093) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:568) ~[spring-beans-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	... 55 common frames omitted
Caused by: java.lang.NoClassDefFoundError: discord4j.core.object.entity.Webhook
	at java.base/java.lang.Class.getDeclaredMethodsImpl(Native Method) ~[na:na]
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:1129) ~[na:na]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:463) ~[spring-core-5.2.8.RELEASE.jar!/:5.2.8.RELEASE]
	... 62 common frames omitted
Caused by: java.lang.ClassNotFoundException: discord4j.core.object.entity.Webhook
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:605) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClassHelper(ClassLoader.java:1180) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1095) ~[na:na]
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:151) ~[aki.jar:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1078) ~[na:na]
	... 65 common frames omitted
```

--- a/src/main/java/club/minnced/discord/webhook/WebhookClient.java
+++ b/src/main/java/club/minnced/discord/webhook/WebhookClient.java
@@ -19,9 +19,10 @@ package club.minnced.discord.webhook;
 import club.minnced.discord.webhook.exception.HttpException;
 import club.minnced.discord.webhook.receive.EntityFactory;
 import club.minnced.discord.webhook.receive.ReadonlyMessage;
-import club.minnced.discord.webhook.send.*;
-import discord4j.core.spec.MessageCreateSpec;
-import net.dv8tion.jda.api.entities.Message;
+import club.minnced.discord.webhook.send.AllowedMentions;
+import club.minnced.discord.webhook.send.WebhookEmbed;
+import club.minnced.discord.webhook.send.WebhookMessage;
+import club.minnced.discord.webhook.send.WebhookMessageBuilder;
 import okhttp3.OkHttpClient;
 import okhttp3.RequestBody;
 import okhttp3.Response;
@@ -31,9 +32,7 @@ import org.json.JSONException;
 import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import reactor.core.publisher.Mono;
 
-import javax.annotation.CheckReturnValue;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
@@ -44,7 +43,6 @@ import java.time.temporal.ChronoUnit;
 import java.util.Collection;
 import java.util.Objects;
 import java.util.concurrent.*;
-import java.util.function.Consumer;
 import java.util.regex.Matcher;
 
 /**
@@ -164,156 +162,6 @@ public class WebhookClient implements AutoCloseable {
     public boolean isShutdown() {
         return isShutdown;
     }
-
-    /////////////////////////////////
-    /// Third-party compatibility ///
-    /////////////////////////////////
-
-    /**
-     * Creates a WebhookClient for the provided webhook.
-     *
-     * @param  webhook
-     *         The webhook
-     *
-     * @throws NullPointerException
-     *         If the webhook is null or does not provide a token
-     *
-     * @return The WebhookClient
-     */
-    @NotNull
-    public static WebhookClient fromJDA(@NotNull net.dv8tion.jda.api.entities.Webhook webhook) {
-        return WebhookClientBuilder.fromJDA(webhook).build();
-    }
-
-    /**
-     * Creates a WebhookClient for the provided webhook.
-     *
-     * @param  webhook
-     *         The webhook
-     *
-     * @throws NullPointerException
-     *         If the webhook is null or does not provide a token
-     *
-     * @return The WebhookClient
-     */
-    @NotNull
-    public static WebhookClient fromD4J(@NotNull discord4j.core.object.entity.Webhook webhook) {
-        return WebhookClientBuilder.fromD4J(webhook).build();
-    }
-
-    /**
-     * Creates a WebhookClient for the provided webhook.
-     *
-     * @param  webhook
-     *         The webhook
-     *
-     * @throws NullPointerException
-     *         If the webhook is null or does not provide a token
-     *
-     * @return The WebhookClient
-     */
-    @NotNull
-    public static WebhookClient fromJavacord(@NotNull org.javacord.api.entity.webhook.Webhook webhook) {
-        return WebhookClientBuilder.fromJavacord(webhook).build();
-    }
-
-    /**
-     * Sends the provided {@link net.dv8tion.jda.api.entities.Message Message} to the webhook.
-     *
-     * @param  message
-     *         The message to send
-     *
-     * @throws NullPointerException
-     *         If null is provided
-     *
-     * @return {@link CompletableFuture}
-     *
-     * @see    #isWait()
-     * @see    WebhookMessageBuilder#fromJDA(Message)
-     */
-    @NotNull
-    public CompletableFuture<ReadonlyMessage> sendJDA(@NotNull net.dv8tion.jda.api.entities.Message message) {
-        return send(WebhookMessageBuilder.fromJDA(message).build());
-    }
-
-    /**
-     * Sends the provided {@link org.javacord.api.entity.message.Message Message} to the webhook.
-     *
-     * @param  message
-     *         The message to send
-     *
-     * @throws NullPointerException
-     *         If null is provided
-     *
-     * @return {@link CompletableFuture}
-     *
-     * @see    #isWait()
-     * @see    WebhookMessageBuilder#fromJavacord(org.javacord.api.entity.message.Message)
-     */
-    @NotNull
-    public CompletableFuture<ReadonlyMessage> sendJavacord(@NotNull org.javacord.api.entity.message.Message message) {
-        return send(WebhookMessageBuilder.fromJavacord(message).build());
-    }
-
-    /**
-     * Sends the provided {@link MessageCreateSpec} to the webhook.
-     *
-     * @param  callback
-     *         The callback used to specify the desired message settings
-     *
-     * @throws NullPointerException
-     *         If null is provided
-     *
-     * @return {@link Mono}
-     *
-     * @see    #isWait()
-     * @see    WebhookMessageBuilder#fromD4J(Consumer)
-     */
-    @NotNull
-    @CheckReturnValue
-    public Mono<ReadonlyMessage> sendD4J(@NotNull Consumer<? super MessageCreateSpec> callback) {
-        WebhookMessage message = WebhookMessageBuilder.fromD4J(callback).build();
-        return Mono.fromFuture(() -> send(message));
-    }
-
-    /**
-     * Sends the provided {@link net.dv8tion.jda.api.entities.MessageEmbed MessageEmbed} to the webhook.
-     *
-     * @param  embed
-     *         The embed to send
-     *
-     * @throws NullPointerException
-     *         If null is provided
-     *
-     * @return {@link CompletableFuture}
-     *
-     * @see    #isWait()
-     * @see    WebhookEmbedBuilder#fromJDA(net.dv8tion.jda.api.entities.MessageEmbed)
-     */
-    @NotNull
-    public CompletableFuture<ReadonlyMessage> sendJDA(@NotNull net.dv8tion.jda.api.entities.MessageEmbed embed) {
-        return send(WebhookEmbedBuilder.fromJDA(embed).build());
-    }
-
-    /**
-     * Sends the provided {@link org.javacord.api.entity.message.embed.Embed Embed} to the webhook.
-     *
-     * @param  embed
-     *         The embed to send
-     *
-     * @throws NullPointerException
-     *         If null is provided
-     *
-     * @return {@link CompletableFuture}
-     *
-     * @see    #isWait()
-     * @see    WebhookEmbedBuilder#fromJavacord(org.javacord.api.entity.message.embed.Embed)
-     */
-    @NotNull
-    public CompletableFuture<ReadonlyMessage> sendJavacord(@NotNull org.javacord.api.entity.message.embed.Embed embed) {
-        return send(WebhookEmbedBuilder.fromJavacord(embed).build());
-    }
-
 
     /**
      * Sends the provided {@link club.minnced.discord.webhook.send.WebhookMessage}

--- a/src/main/java/club/minnced/discord/webhook/WebhookClientBuilder.java
+++ b/src/main/java/club/minnced/discord/webhook/WebhookClientBuilder.java
@@ -16,6 +16,9 @@
 
 package club.minnced.discord.webhook;
 
+import club.minnced.discord.webhook.external.D4JWebhookClient;
+import club.minnced.discord.webhook.external.JDAWebhookClient;
+import club.minnced.discord.webhook.external.JavacordWebhookClient;
 import club.minnced.discord.webhook.send.AllowedMentions;
 import okhttp3.OkHttpClient;
 import org.jetbrains.annotations.NotNull;
@@ -256,6 +259,45 @@ public class WebhookClientBuilder { //TODO: tests
         OkHttpClient client = this.client == null ? new OkHttpClient() : this.client;
         ScheduledExecutorService pool = this.pool != null ? this.pool : getDefaultPool(id, threadFactory, isDaemon);
         return new WebhookClient(id, token, parseMessage, client, pool, allowedMentions);
+    }
+
+    /**
+     * Builds the {@link club.minnced.discord.webhook.external.JDAWebhookClient}
+     * with the current settings
+     *
+     * @return {@link club.minnced.discord.webhook.external.JDAWebhookClient} instance
+     */
+    @NotNull
+    public JDAWebhookClient buildJDA() {
+        OkHttpClient client = this.client == null ? new OkHttpClient() : this.client;
+        ScheduledExecutorService pool = this.pool != null ? this.pool : getDefaultPool(id, threadFactory, isDaemon);
+        return new JDAWebhookClient(id, token, parseMessage, client, pool, allowedMentions);
+    }
+
+    /**
+     * Builds the {@link club.minnced.discord.webhook.external.D4JWebhookClient}
+     * with the current settings
+     *
+     * @return {@link club.minnced.discord.webhook.external.D4JWebhookClient} instance
+     */
+    @NotNull
+    public D4JWebhookClient buildD4J() {
+        OkHttpClient client = this.client == null ? new OkHttpClient() : this.client;
+        ScheduledExecutorService pool = this.pool != null ? this.pool : getDefaultPool(id, threadFactory, isDaemon);
+        return new D4JWebhookClient(id, token, parseMessage, client, pool, allowedMentions);
+    }
+
+    /**
+     * Builds the {@link club.minnced.discord.webhook.external.JavacordWebhookClient}
+     * with the current settings
+     *
+     * @return {@link club.minnced.discord.webhook.external.JavacordWebhookClient} instance
+     */
+    @NotNull
+    public JavacordWebhookClient buildJavacord() {
+        OkHttpClient client = this.client == null ? new OkHttpClient() : this.client;
+        ScheduledExecutorService pool = this.pool != null ? this.pool : getDefaultPool(id, threadFactory, isDaemon);
+        return new JavacordWebhookClient(id, token, parseMessage, client, pool, allowedMentions);
     }
 
     protected static ScheduledExecutorService getDefaultPool(long id, ThreadFactory factory, boolean isDaemon) {

--- a/src/main/java/club/minnced/discord/webhook/external/D4JWebhookClient.java
+++ b/src/main/java/club/minnced/discord/webhook/external/D4JWebhookClient.java
@@ -1,0 +1,59 @@
+package club.minnced.discord.webhook.external;
+
+import club.minnced.discord.webhook.WebhookClient;
+import club.minnced.discord.webhook.WebhookClientBuilder;
+import club.minnced.discord.webhook.receive.ReadonlyMessage;
+import club.minnced.discord.webhook.send.AllowedMentions;
+import club.minnced.discord.webhook.send.WebhookMessage;
+import club.minnced.discord.webhook.send.WebhookMessageBuilder;
+import discord4j.core.spec.MessageCreateSpec;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+import reactor.core.publisher.Mono;
+
+import javax.annotation.CheckReturnValue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Consumer;
+
+public class D4JWebhookClient extends WebhookClient {
+    public D4JWebhookClient(long id, String token, boolean parseMessage, OkHttpClient client, ScheduledExecutorService pool, AllowedMentions mentions) {
+        super(id, token, parseMessage, client, pool, mentions);
+    }
+
+    /**
+     * Creates a WebhookClient for the provided webhook.
+     *
+     * @param  webhook
+     *         The webhook
+     *
+     * @throws NullPointerException
+     *         If the webhook is null or does not provide a token
+     *
+     * @return The WebhookClient
+     */
+    @NotNull
+    public static WebhookClient from(@NotNull discord4j.core.object.entity.Webhook webhook) {
+        return WebhookClientBuilder.fromD4J(webhook).build();
+    }
+
+    /**
+     * Sends the provided {@link MessageCreateSpec} to the webhook.
+     *
+     * @param  callback
+     *         The callback used to specify the desired message settings
+     *
+     * @throws NullPointerException
+     *         If null is provided
+     *
+     * @return {@link Mono}
+     *
+     * @see    #isWait()
+     * @see    WebhookMessageBuilder#fromD4J(Consumer)
+     */
+    @NotNull
+    @CheckReturnValue
+    public Mono<ReadonlyMessage> send(@NotNull Consumer<? super MessageCreateSpec> callback) {
+        WebhookMessage message = WebhookMessageBuilder.fromD4J(callback).build();
+        return Mono.fromFuture(() -> send(message));
+    }
+}

--- a/src/main/java/club/minnced/discord/webhook/external/JDAWebhookClient.java
+++ b/src/main/java/club/minnced/discord/webhook/external/JDAWebhookClient.java
@@ -1,0 +1,65 @@
+package club.minnced.discord.webhook.external;
+
+import club.minnced.discord.webhook.WebhookClient;
+import club.minnced.discord.webhook.WebhookClientBuilder;
+import club.minnced.discord.webhook.receive.ReadonlyMessage;
+import club.minnced.discord.webhook.send.AllowedMentions;
+import club.minnced.discord.webhook.send.WebhookEmbedBuilder;
+import club.minnced.discord.webhook.send.WebhookMessageBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class JDAWebhookClient extends WebhookClient {
+    public JDAWebhookClient(long id, String token, boolean parseMessage, OkHttpClient client, ScheduledExecutorService pool, AllowedMentions mentions) {
+        super(id, token, parseMessage, client, pool, mentions);
+    }
+
+    /**
+     * Creates a WebhookClient for the provided webhook.
+     *
+     * @param webhook The webhook
+     * @return The WebhookClient
+     * @throws NullPointerException If the webhook is null or does not provide a token
+     */
+    @NotNull
+    public static WebhookClient fromJDA(@NotNull net.dv8tion.jda.api.entities.Webhook webhook) {
+        return WebhookClientBuilder.fromJDA(webhook).build();
+    }
+
+    /**
+     * Sends the provided {@link net.dv8tion.jda.api.entities.Message Message} to the webhook.
+     *
+     * @param  message
+     *         The message to send
+     *
+     * @throws NullPointerException
+     *         If null is provided
+     *
+     * @return {@link CompletableFuture}
+     *
+     * @see    #isWait()
+     * @see    WebhookMessageBuilder#fromJDA(Message)
+     */
+    @NotNull
+    public CompletableFuture<ReadonlyMessage> send(@NotNull net.dv8tion.jda.api.entities.Message message) {
+        return send(WebhookMessageBuilder.fromJDA(message).build());
+    }
+
+    /**
+     * Sends the provided {@link net.dv8tion.jda.api.entities.MessageEmbed MessageEmbed} to the webhook.
+     *
+     * @param embed The embed to send
+     * @return {@link CompletableFuture}
+     * @throws NullPointerException If null is provided
+     * @see #isWait()
+     * @see WebhookEmbedBuilder#fromJDA(net.dv8tion.jda.api.entities.MessageEmbed)
+     */
+    @NotNull
+    public CompletableFuture<ReadonlyMessage> send(@NotNull net.dv8tion.jda.api.entities.MessageEmbed embed) {
+        return send(WebhookEmbedBuilder.fromJDA(embed).build());
+    }
+}

--- a/src/main/java/club/minnced/discord/webhook/external/JavacordWebhookClient.java
+++ b/src/main/java/club/minnced/discord/webhook/external/JavacordWebhookClient.java
@@ -1,0 +1,73 @@
+package club.minnced.discord.webhook.external;
+
+import club.minnced.discord.webhook.WebhookClient;
+import club.minnced.discord.webhook.WebhookClientBuilder;
+import club.minnced.discord.webhook.receive.ReadonlyMessage;
+import club.minnced.discord.webhook.send.AllowedMentions;
+import club.minnced.discord.webhook.send.WebhookEmbedBuilder;
+import club.minnced.discord.webhook.send.WebhookMessageBuilder;
+import okhttp3.OkHttpClient;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class JavacordWebhookClient extends WebhookClient {
+    public JavacordWebhookClient(long id, String token, boolean parseMessage, OkHttpClient client, ScheduledExecutorService pool, AllowedMentions mentions) {
+        super(id, token, parseMessage, client, pool, mentions);
+    }
+
+    /**
+     * Creates a WebhookClient for the provided webhook.
+     *
+     * @param  webhook
+     *         The webhook
+     *
+     * @throws NullPointerException
+     *         If the webhook is null or does not provide a token
+     *
+     * @return The WebhookClient
+     */
+    @NotNull
+    public static WebhookClient from(@NotNull org.javacord.api.entity.webhook.Webhook webhook) {
+        return WebhookClientBuilder.fromJavacord(webhook).build();
+    }
+
+    /**
+     * Sends the provided {@link org.javacord.api.entity.message.Message Message} to the webhook.
+     *
+     * @param  message
+     *         The message to send
+     *
+     * @throws NullPointerException
+     *         If null is provided
+     *
+     * @return {@link CompletableFuture}
+     *
+     * @see    #isWait()
+     * @see    WebhookMessageBuilder#fromJavacord(org.javacord.api.entity.message.Message)
+     */
+    @NotNull
+    public CompletableFuture<ReadonlyMessage> send(@NotNull org.javacord.api.entity.message.Message message) {
+        return send(WebhookMessageBuilder.fromJavacord(message).build());
+    }
+
+    /**
+     * Sends the provided {@link org.javacord.api.entity.message.embed.Embed Embed} to the webhook.
+     *
+     * @param  embed
+     *         The embed to send
+     *
+     * @throws NullPointerException
+     *         If null is provided
+     *
+     * @return {@link CompletableFuture}
+     *
+     * @see    #isWait()
+     * @see    WebhookEmbedBuilder#fromJavacord(org.javacord.api.entity.message.embed.Embed)
+     */
+    @NotNull
+    public CompletableFuture<ReadonlyMessage> send(@NotNull org.javacord.api.entity.message.embed.Embed embed) {
+        return send(WebhookEmbedBuilder.fromJavacord(embed).build());
+    }
+}

